pkg/allocator/allocator.go

```diff
@@ -178,8 +181,9 @@ type CacheValidator func(kind AllocatorChangeKind, id idpool.ID, key AllocatorKe
 // The allocator can be configured by passing in additional options:
 //   - WithMin(id) - minimum ID to allocate (default: 1)
 //   - WithMax(id) - maximum ID to allocate (default max(uint64))
-func NewAllocatorForGC(backend Backend, opts ...AllocatorOption) *Allocator {
+func NewAllocatorForGC(rootLogger *slog.Logger, backend Backend, opts ...AllocatorOption) *Allocator {
 	a := &Allocator{
+		logger:  rootLogger.With(subsysLogAttr),
```

Comment:
What does it happen if the root logger already defines a `subsys` attribute? I assume that is the case for the one passed in `kvstore_gc.go`.

Reply:
Right now it's harmless because I haven't pushed some changes I have locally to detect this.

In the future I'm planning to panic the agent if it has any duplicated keys when running in CI.

For the `kvstore_gc.go` case, it doesn't have any subsys. All `slog.Logger` created by the hive have a "module" attribute instead.

Reply:
Ah got it, I thought hive was also using `subsys` as key. All good then.